pyproject.toml

```diff
@@ -14,147 +14,133 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-[tool.poetry]
+[project]
 name = "pyiceberg"
 version = "0.10.0"
-readme = "README.md"
-homepage = "https://py.iceberg.apache.org/"
-repository = "https://github.com/apache/iceberg-python"
 description = "Apache Iceberg is an open table format for huge analytic datasets"
-authors = ["Apache Software Foundation <dev@iceberg.apache.org>"]
-license = "Apache License 2.0"
+authors = [{ name = "Apache Software Foundation", email = "dev@iceberg.apache.org" }]
+requires-python = ">=3.10.0,<4.0.0"
+readme = "README.md"
+license = "Apache-2.0"
+license-files = ["LICEN[CS]E*", "NOTICE*"]
 classifiers = [
-  "License :: OSI Approved :: Apache Software License",
-  "Operating System :: OS Independent",
-  "Programming Language :: Python :: 3.10",
-  "Programming Language :: Python :: 3.11",
-  "Programming Language :: Python :: 3.12",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+]
+dependencies = [
+    "mmh3>=4.0.0,<6.0.0",
+    "requests>=2.20.0,<3.0.0",
+    "click>=7.1.1,<9.0.0",
+    "rich>=10.11.0,<15.0.0",
+    "strictyaml>=1.7.0,<2.0.0", # CVE-2020-14343 was fixed in 5.4.
+    "pydantic>=2.0,<3.0,!=2.4.0,!=2.4.1,!=2.12.0,!=2.12.1", # 2.4.0, 2.4.1, 2.12.0, 2.12.1 has a critical bug
+    "sortedcontainers==2.4.0",
+    "fsspec>=2023.1.0",
+    "pyparsing>=3.1.0,<4.0.0",
+    "tenacity>=8.2.3,<10.0.0",
+    "pyroaring>=1.0.0,<2.0.0",
+    "cachetools>=5.5,<7.0",
+    "zstandard>=0.13.0,<1.0.0"
+]
+
+[project.optional-dependencies]
+pyarrow = [
+    "pyarrow>=17.0.0",
+    "pyiceberg-core>=0.5.1,<0.8.0",
+]
+pandas = [
+    "pyiceberg[pyarrow]",
+    "pandas>=1.0.0,<3.0.0",
+]
+duckdb = [
+    "pyiceberg[pyarrow]",
+    "duckdb>=0.5.0,<2.0.0",
+]
+ray = [
+    "pyiceberg[pyarrow]",
```

Comment:
nit: i think `pyiceberg[pyarrow]` here will install `pyiceberg-core` as well. 

For correctness, i think it might be better to go back to using `pyarrow>=17.0.0`. Even with the duplication, i think uv will edit all the instances at once